### ISSUE TITLE
Wrong bindings

### ISSUE THREAD

**Issue by listepo** [dart-lang/ffigen#71]
In my case:

from
```c
typedef void (*PQnoticeProcessor) (void *arg, const char *message);

extern PQnoticeProcessor PQsetNoticeProcessor(PQnoticeProcessor proc, void *arg);
```
to
```dart
/// AUTO GENERATED FILE, DO NOT EDIT.
///
/// Generated by `package:ffigen`.
import 'dart:ffi' as ffi;

class PG {
  /// Holds the Dynamic library.
  final ffi.DynamicLibrary _dylib;

  /// The symbols are looked up in [dynamicLibrary].
  PG(ffi.DynamicLibrary dynamicLibrary) : _dylib = dynamicLibrary;

  ffi.Pointer<ffi.NativeFunction<PQnoticeProcessor_1>> PQsetNoticeProcessor(
    ffi.Pointer<ffi.NativeFunction<PQnoticeProcessor_2>> proc,
    ffi.Pointer<ffi.Void> arg,
  ) {
    _PQsetNoticeProcessor ??= _dylib.lookupFunction<_c_PQsetNoticeProcessor,
        _dart_PQsetNoticeProcessor>('PQsetNoticeProcessor');
    return _PQsetNoticeProcessor(
      proc,
      arg,
    );
  }

  _dart_PQsetNoticeProcessor _PQsetNoticeProcessor;
}

typedef PQnoticeProcessor_2 = ffi.Void Function(
  ffi.Pointer<ffi.Void>,
  ffi.Pointer<ffi.Int8>,
);

typedef _c_PQsetNoticeProcessor
    = ffi.Pointer<ffi.NativeFunction<PQnoticeProcessor_1>> Function(
  ffi.Pointer<ffi.NativeFunction<PQnoticeProcessor_2>> proc,
  ffi.Pointer<ffi.Void> arg,
);

typedef _dart_PQsetNoticeProcessor
    = ffi.Pointer<ffi.NativeFunction<PQnoticeProcessor_1>> Function(
  ffi.Pointer<ffi.NativeFunction<PQnoticeProcessor_2>> proc,
  ffi.Pointer<ffi.Void> arg,
);

```

I guess  `PQnoticeProcessor_1` and `PQnoticeProcessor_2` should be `PQnoticeProcessor`

**Comment by mannprerak2** [comment]
Thank you for filing this issue.
Currently, we treat every function typedef seperately (so if a function typedef is used a 100 times, it will be generated a 100 times) but I suppose we can reuse typedefs having same names and types.

P.S Just want to confirm, is the typedef `PQnoticeProcessor_1` not being generated?

**Comment by listepo** [comment]
@mannprerak2 yes, not being generated.

**Comment by listepo** [comment]
this is a complete example, I. have only one `typedef void (*PQnoticeProcessor) (void *arg, const char *message);`, but after `pub run ffigen` I got PQnoticeProcessor_1 and PQnoticeProcessor_2. Only PQnoticeProcessor_2 with `typedef`

**Comment by dcharkes** [comment]
This looks like two issues:

1. The generated code doesn't work (which could be fixed by generating the missing typedef)
2. The generated code should generate a public typedef which is reused (which should probably be addressed with #24).
